### DISCUSSION TITLE
用户资源与前端权限问题

### DISCUSSION THREAD

**Discussion by hunao0221** [zuihou/zuihou-admin-cloud#306] [企业版]
```java
# ResourceBiz#findVisibleResource

List<String> resourceCodes = Arrays.asList(ResourceTypeEnum.BUTTON.getCode(), ResourceTypeEnum.VIEW.getCode(), ResourceTypeEnum.FIELD.getCode());
```
查询用户资源的接口排除了菜单类型的资源，这是为什么？ 一般在菜单上不是就关联了相关的接口了吗（比如page, tree）
假设一个场景，用户点击菜单1中的列表的item1按钮，跳转到菜单2（查看与item相关联的数据），这个按钮显示的条件是该用户拥有菜单2的资源。由于后端接口不会返回菜单类型的资源，前端在加auth条件的话，就需要再在这个菜单下新增查看的功能了，该功能仍然需要关联page, tree 这样的接口才行，有点冗余。

**Comment by zuihou** [comment]
# ResourceBiz#findVisibleResource  
这个方法只是前端控制 按钮权限（用户能不能看到某个页面上的按钮） 的。

# ResourceBiz#findVisibleRouter
这个接口用于获取 用户的菜单和隐藏菜单的。
